frontend/src/modules/search/components/FiltersPanel.js

```diff
@@ -73,6 +231,16 @@ export class FiltersPanelBase extends React.Component<Props, State> {
         return this.props.update();
     }
 
+    createToggleTimeRangeFilter() {
+        const { chartFrom, chartTo, isChartVisible } = this.state;
+
+        if (isChartVisible) {
+            return;
+        }
+
+        return this.createToggleFilter([chartFrom, chartTo].join('-'), 'timeRange');
+    }
```

Comment:
That is a weird construct that is not required. The reason to make a method that returns a function is to create a closure that preserves a variable. In this case, you don't need that, because your variable is in `state`. So you could here instead create a `toggleTimeRangeFilter` method that combines the content of this method and of `createToggleFilter`'s returned function.